In [6]:
# YOLOv8 Training Script

from ultralytics import YOLO
import os

# Load a pretrained YOLOv8n variant (custom lightweight model)
model = YOLO('../src/human_detector/models/yolov11n.pt')


# Training with simple parameters
#    - data: path to your data.yaml
#    - epochs, imgsz, project, name: same CLI args
# results = model.train(
#     data    = 'data.yaml',
#     epochs  = 1,
#     imgsz   = 640,
#     project = 'human_robot',
#     name    = 'exp1'
# )


# Train the model with inline arguments
results = model.train(
    data='data.yaml',
    epochs=120,
    patience=20,
    imgsz=416,
    batch=2,
    device=0,
    half=True,
    workers=2,
    cache=False,
    
    # Low-light augmentation settings
    hsv_h=0.015,
    hsv_s=0.7,
    hsv_v=0.4,
    translate=0.1,
    scale=0.5,
    fliplr=0.5,
    mosaic=1.0,
    mixup=0.0,
    copy_paste=0.0
)

# Save trained model path
save_dir = str(results.save_dir)
best_pt = os.path.join(save_dir, 'weights', 'best.pt')
print(f"✅  Trained checkpoint saved to: {best_pt}")


Ultralytics 8.3.141 🚀 Python-3.10.12 torch-2.7.0+cu126 CUDA:0 (NVIDIA GeForce MX550, 1864MiB)
engine/trainer: agnostic_nms=False, amp=True, augment=False, auto_augment=randaugment, batch=2, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=data.yaml, degrees=0.0, deterministic=True, device=0, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, epochs=120, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=True, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=416, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=../src/human_detector/models/yolov11n.pt, momentum=0.937, mosaic=1.0, multi_scale=False, name=train2, nbs=64, nms=False, opset=None, optimize=False, optimizer=auto, overlap_mask=True, patience=20, perspective=0.0, plots=True, pos

train: Scanning /home/yogee/Desktop/human_detector_ws/training_data/labels/train.cache... 1324 images, 0 backgrounds, 0 corrupt: 100%|██████████| 1324/1324 [00:00<?, ?it/s]


val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 4533.1±3502.4 MB/s, size: 335.7 KB)


val: Scanning /home/yogee/Desktop/human_detector_ws/training_data/labels/val.cache... 331 images, 0 backgrounds, 0 corrupt: 100%|██████████| 331/331 [00:00<?, ?it/s]


Plotting labels to runs/detect/train2/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.001667, momentum=0.9) with parameter groups 81 weight(decay=0.0), 88 weight(decay=0.0005), 87 bias(decay=0.0)
Image sizes 416 train, 416 val
Using 2 dataloader workers
Logging results to runs/detect/train2
Starting training for 120 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      1/120     0.334G     0.9983      1.404      1.098         15        416: 100%|██████████| 662/662 [00:54<00:00, 12.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 24.74it/s]


                   all        331       1034      0.953      0.932      0.961      0.703

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      2/120     0.352G     0.9137     0.8635      1.064         13        416: 100%|██████████| 662/662 [00:56<00:00, 11.79it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.19it/s]

                   all        331       1034      0.965       0.94      0.969      0.761



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      3/120     0.365G     0.8728     0.7578      1.048         12        416: 100%|██████████| 662/662 [00:58<00:00, 11.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 20.90it/s]

                   all        331       1034      0.967      0.943      0.972      0.775



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      4/120     0.381G     0.8434     0.6969      1.044         18        416: 100%|██████████| 662/662 [00:58<00:00, 11.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.14it/s]

                   all        331       1034      0.968       0.97      0.974      0.799



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      5/120     0.395G     0.8146       0.65       1.03          7        416: 100%|██████████| 662/662 [00:57<00:00, 11.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.71it/s]

                   all        331       1034      0.949      0.975      0.969        0.8



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      6/120     0.408G     0.7836     0.5968      1.018         12        416: 100%|██████████| 662/662 [00:58<00:00, 11.25it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 20.93it/s]

                   all        331       1034      0.959      0.975      0.974      0.812



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      7/120     0.424G     0.7616     0.5763      1.009         16        416: 100%|██████████| 662/662 [00:58<00:00, 11.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.44it/s]

                   all        331       1034      0.973      0.965      0.981      0.822



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      8/120     0.424G     0.7232     0.5495     0.9959          6        416: 100%|██████████| 662/662 [00:58<00:00, 11.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.94it/s]

                   all        331       1034      0.969      0.974      0.981      0.831



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      9/120     0.424G     0.7205     0.5275     0.9949         20        416: 100%|██████████| 662/662 [00:59<00:00, 11.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 20.77it/s]

                   all        331       1034      0.965      0.976      0.985      0.854



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     10/120     0.424G     0.7027     0.5208     0.9846          7        416: 100%|██████████| 662/662 [00:58<00:00, 11.35it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.31it/s]

                   all        331       1034      0.971      0.971      0.982      0.841



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     11/120     0.424G     0.6968     0.5122     0.9891         17        416: 100%|██████████| 662/662 [00:57<00:00, 11.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.67it/s]

                   all        331       1034      0.974      0.963      0.983      0.845



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     12/120     0.424G     0.6965     0.5135     0.9872         12        416: 100%|██████████| 662/662 [00:58<00:00, 11.25it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 20.83it/s]

                   all        331       1034      0.976      0.965      0.983      0.843



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     13/120     0.424G     0.6668     0.4891     0.9728         14        416: 100%|██████████| 662/662 [00:58<00:00, 11.41it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.49it/s]

                   all        331       1034      0.972      0.969      0.984       0.87



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     14/120     0.424G     0.6718     0.4754     0.9758         14        416: 100%|██████████| 662/662 [00:57<00:00, 11.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.85it/s]

                   all        331       1034      0.972      0.974      0.985      0.876



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     15/120     0.424G     0.6591     0.4719      0.974          3        416: 100%|██████████| 662/662 [00:58<00:00, 11.22it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.03it/s]

                   all        331       1034      0.964      0.976      0.985      0.852



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     16/120     0.424G     0.6429     0.4617     0.9656         10        416: 100%|██████████| 662/662 [00:58<00:00, 11.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.39it/s]

                   all        331       1034      0.977      0.967      0.984      0.875



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     17/120     0.424G      0.645     0.4587     0.9717         18        416: 100%|██████████| 662/662 [00:58<00:00, 11.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.03it/s]

                   all        331       1034      0.962      0.974      0.987      0.875



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     18/120     0.424G     0.6373      0.449     0.9647          9        416: 100%|██████████| 662/662 [00:58<00:00, 11.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.24it/s]

                   all        331       1034       0.97      0.976      0.986      0.884



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     19/120     0.424G     0.6286     0.4407     0.9653         14        416: 100%|██████████| 662/662 [00:59<00:00, 11.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.50it/s]

                   all        331       1034      0.973      0.969      0.985       0.88



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     20/120     0.438G     0.6289      0.436     0.9626          6        416: 100%|██████████| 662/662 [00:57<00:00, 11.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.03it/s]

                   all        331       1034      0.966      0.978      0.987      0.883



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     21/120     0.438G     0.6265     0.4341     0.9572          8        416: 100%|██████████| 662/662 [00:58<00:00, 11.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.88it/s]

                   all        331       1034      0.971      0.976      0.987      0.888



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     22/120     0.438G     0.6113     0.4244     0.9489         16        416: 100%|██████████| 662/662 [00:59<00:00, 11.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.11it/s]

                   all        331       1034       0.97      0.974      0.985      0.875



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     23/120     0.438G     0.6009     0.4163     0.9492          8        416: 100%|██████████| 662/662 [00:58<00:00, 11.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.51it/s]

                   all        331       1034      0.976      0.969      0.987      0.898



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     24/120     0.438G     0.5974     0.4153     0.9522          8        416: 100%|██████████| 662/662 [00:58<00:00, 11.41it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.15it/s]

                   all        331       1034      0.972      0.968      0.986      0.883



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     25/120     0.438G     0.5901     0.4141     0.9425         13        416: 100%|██████████| 662/662 [00:58<00:00, 11.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.42it/s]

                   all        331       1034      0.975      0.971      0.989      0.887



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     26/120     0.438G     0.5855     0.4132     0.9465          9        416: 100%|██████████| 662/662 [00:58<00:00, 11.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.67it/s]

                   all        331       1034      0.973      0.968      0.984      0.887



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     27/120     0.438G     0.5876     0.4102     0.9454         13        416: 100%|██████████| 662/662 [00:57<00:00, 11.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.36it/s]

                   all        331       1034      0.974      0.965      0.987      0.893



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     28/120     0.438G     0.5761     0.4057     0.9427         11        416: 100%|██████████| 662/662 [00:58<00:00, 11.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.30it/s]

                   all        331       1034      0.971      0.973      0.986      0.886



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     29/120     0.438G      0.564     0.3931     0.9365         16        416: 100%|██████████| 662/662 [00:59<00:00, 11.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 20.92it/s]

                   all        331       1034      0.976      0.966      0.987      0.897



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     30/120     0.438G     0.5703     0.3978     0.9388         13        416: 100%|██████████| 662/662 [00:57<00:00, 11.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.24it/s]

                   all        331       1034      0.976      0.968      0.986      0.896



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     31/120     0.438G     0.5645     0.3915     0.9397          7        416: 100%|██████████| 662/662 [00:57<00:00, 11.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.60it/s]

                   all        331       1034      0.967      0.971      0.987      0.904



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     32/120     0.438G      0.579      0.397     0.9454         10        416: 100%|██████████| 662/662 [00:58<00:00, 11.26it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 20.77it/s]

                   all        331       1034      0.971      0.977      0.987      0.909



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     33/120     0.438G     0.5591     0.3885     0.9356          9        416: 100%|██████████| 662/662 [00:58<00:00, 11.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.24it/s]

                   all        331       1034      0.974       0.97      0.987      0.902



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     34/120     0.438G      0.569     0.3925     0.9406         10        416: 100%|██████████| 662/662 [00:58<00:00, 11.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.55it/s]

                   all        331       1034      0.981      0.968      0.988      0.902



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     35/120     0.438G     0.5698     0.3932     0.9438         11        416: 100%|██████████| 662/662 [00:58<00:00, 11.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.08it/s]

                   all        331       1034      0.975      0.977      0.988      0.896



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     36/120     0.438G     0.5435     0.3818     0.9268         20        416: 100%|██████████| 662/662 [00:59<00:00, 11.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.59it/s]

                   all        331       1034       0.98      0.967      0.987      0.901



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     37/120     0.438G     0.5535      0.385     0.9255          4        416: 100%|██████████| 662/662 [00:57<00:00, 11.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.94it/s]

                   all        331       1034      0.981      0.965      0.987      0.908



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     38/120     0.438G     0.5331     0.3729     0.9282          6        416: 100%|██████████| 662/662 [00:58<00:00, 11.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.22it/s]

                   all        331       1034      0.973       0.97      0.987      0.914



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     39/120     0.438G     0.5421     0.3734      0.927         14        416: 100%|██████████| 662/662 [00:59<00:00, 11.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.14it/s]

                   all        331       1034      0.981      0.967      0.989      0.912



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     40/120     0.438G     0.5342     0.3742     0.9293         13        416: 100%|██████████| 662/662 [00:58<00:00, 11.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.04it/s]

                   all        331       1034      0.968      0.973      0.987      0.908



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     41/120     0.438G     0.5285     0.3663      0.927         11        416: 100%|██████████| 662/662 [00:57<00:00, 11.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.26it/s]

                   all        331       1034      0.978      0.966      0.989       0.92



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     42/120     0.438G     0.5418     0.3736     0.9278          7        416: 100%|██████████| 662/662 [00:58<00:00, 11.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:04<00:00, 20.14it/s]

                   all        331       1034      0.977      0.974      0.988       0.91



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     43/120     0.438G     0.5334      0.365     0.9243          7        416: 100%|██████████| 662/662 [00:59<00:00, 11.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.61it/s]

                   all        331       1034       0.98      0.968      0.988      0.915



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     44/120     0.438G     0.5287     0.3694     0.9252         17        416: 100%|██████████| 662/662 [00:58<00:00, 11.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.46it/s]

                   all        331       1034      0.983      0.968      0.988      0.916



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     45/120     0.438G     0.5292     0.3646      0.929         22        416: 100%|██████████| 662/662 [00:59<00:00, 11.16it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.38it/s]

                   all        331       1034      0.978      0.969      0.987      0.913



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     46/120     0.438G     0.5231     0.3611       0.92         14        416: 100%|██████████| 662/662 [00:57<00:00, 11.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.24it/s]

                   all        331       1034       0.98      0.965      0.987      0.905



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     47/120     0.438G       0.51     0.3628     0.9147         18        416: 100%|██████████| 662/662 [00:59<00:00, 11.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 20.93it/s]

                   all        331       1034      0.974      0.971      0.986      0.892



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     48/120     0.438G     0.5177     0.3631     0.9211         15        416: 100%|██████████| 662/662 [00:58<00:00, 11.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.50it/s]

                   all        331       1034      0.977      0.971      0.985      0.913



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     49/120     0.438G     0.5137     0.3529     0.9194         17        416: 100%|██████████| 662/662 [00:57<00:00, 11.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.70it/s]

                   all        331       1034      0.979      0.972       0.99      0.928



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     50/120     0.438G     0.5125     0.3602     0.9218         13        416: 100%|██████████| 662/662 [00:57<00:00, 11.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.82it/s]

                   all        331       1034      0.982      0.967       0.99      0.928



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     51/120     0.438G     0.5071     0.3553     0.9114         15        416: 100%|██████████| 662/662 [00:56<00:00, 11.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.48it/s]

                   all        331       1034      0.976      0.973       0.99      0.922



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     52/120     0.438G     0.5046     0.3458     0.9144         13        416: 100%|██████████| 662/662 [00:57<00:00, 11.57it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.64it/s]

                   all        331       1034      0.972      0.972      0.989      0.921



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     53/120     0.438G     0.5022     0.3462     0.9113         12        416: 100%|██████████| 662/662 [00:57<00:00, 11.41it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.81it/s]

                   all        331       1034      0.976      0.974      0.989      0.929



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     54/120     0.438G     0.4947     0.3443     0.9043         11        416: 100%|██████████| 662/662 [00:57<00:00, 11.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.52it/s]

                   all        331       1034      0.974      0.973      0.989      0.918



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     55/120     0.438G     0.4935     0.3491     0.9055         14        416: 100%|██████████| 662/662 [00:57<00:00, 11.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.06it/s]

                   all        331       1034      0.973      0.975      0.989      0.925



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     56/120     0.438G     0.4918     0.3428     0.9098         14        416: 100%|██████████| 662/662 [00:57<00:00, 11.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.59it/s]

                   all        331       1034      0.971       0.98      0.989      0.918



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     57/120     0.438G     0.4815     0.3368      0.901         16        416: 100%|██████████| 662/662 [00:57<00:00, 11.59it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.70it/s]

                   all        331       1034      0.975      0.972      0.989      0.939



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     58/120     0.438G     0.4847     0.3348     0.9038         13        416: 100%|██████████| 662/662 [00:57<00:00, 11.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.35it/s]

                   all        331       1034      0.972      0.979      0.989      0.926



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     59/120     0.438G     0.4906     0.3372     0.9091          6        416: 100%|██████████| 662/662 [00:58<00:00, 11.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.77it/s]

                   all        331       1034      0.978      0.977      0.989      0.922



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     60/120     0.438G     0.4796     0.3356      0.902         17        416: 100%|██████████| 662/662 [00:58<00:00, 11.41it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.07it/s]

                   all        331       1034      0.982      0.971      0.989       0.93



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     61/120     0.438G     0.4862     0.3393     0.9141          9        416: 100%|██████████| 662/662 [00:57<00:00, 11.60it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.80it/s]

                   all        331       1034      0.978      0.974      0.988      0.932



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     62/120     0.438G      0.488     0.3372     0.9117          6        416: 100%|██████████| 662/662 [00:57<00:00, 11.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.95it/s]

                   all        331       1034      0.973      0.977      0.992      0.929



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     63/120     0.438G     0.4736     0.3304     0.9003         19        416: 100%|██████████| 662/662 [00:57<00:00, 11.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.32it/s]

                   all        331       1034      0.988      0.959      0.989      0.926



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     64/120     0.438G     0.4729     0.3304     0.9068          8        416: 100%|██████████| 662/662 [00:59<00:00, 11.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:04<00:00, 20.64it/s]

                   all        331       1034      0.976      0.973      0.989      0.918



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     65/120     0.438G     0.4747     0.3372     0.9066         10        416: 100%|██████████| 662/662 [00:59<00:00, 11.12it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 20.88it/s]

                   all        331       1034      0.975      0.969      0.988      0.929



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     66/120     0.438G     0.4833     0.3362     0.9093          8        416: 100%|██████████| 662/662 [00:59<00:00, 11.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.86it/s]

                   all        331       1034      0.975      0.976       0.99      0.931



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     67/120     0.438G     0.4682     0.3291     0.9023         13        416: 100%|██████████| 662/662 [00:58<00:00, 11.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.47it/s]

                   all        331       1034      0.977      0.972      0.989       0.93



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     68/120     0.438G     0.4626     0.3214     0.9039         16        416: 100%|██████████| 662/662 [00:59<00:00, 11.14it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.11it/s]

                   all        331       1034      0.977      0.972      0.989      0.934



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     69/120     0.438G     0.4613     0.3251     0.9003          5        416: 100%|██████████| 662/662 [00:59<00:00, 11.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:04<00:00, 20.43it/s]

                   all        331       1034      0.977      0.975      0.989      0.939



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     70/120     0.438G     0.4642     0.3269     0.9013         10        416: 100%|██████████| 662/662 [00:58<00:00, 11.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.79it/s]

                   all        331       1034      0.979      0.969      0.989      0.936



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     71/120     0.438G     0.4626      0.324     0.8998          5        416: 100%|██████████| 662/662 [00:58<00:00, 11.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.08it/s]

                   all        331       1034      0.979      0.977      0.989      0.937



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     72/120     0.438G      0.462     0.3216      0.906          8        416: 100%|██████████| 662/662 [00:59<00:00, 11.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:04<00:00, 20.65it/s]

                   all        331       1034      0.978      0.973      0.989       0.94



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     73/120     0.438G     0.4626     0.3222     0.9017          8        416: 100%|██████████| 662/662 [00:58<00:00, 11.25it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.18it/s]

                   all        331       1034      0.976      0.978      0.989      0.942



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     74/120     0.438G     0.4606     0.3203     0.9003         11        416: 100%|██████████| 662/662 [00:59<00:00, 11.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:04<00:00, 20.72it/s]

                   all        331       1034       0.98      0.973      0.989      0.938



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     75/120     0.438G     0.4387     0.3159     0.8916         10        416: 100%|██████████| 662/662 [00:59<00:00, 11.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.17it/s]

                   all        331       1034      0.978      0.972       0.99      0.935



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     76/120     0.438G     0.4477     0.3086     0.8928         17        416: 100%|██████████| 662/662 [00:58<00:00, 11.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.53it/s]

                   all        331       1034      0.975      0.977      0.991      0.942



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     77/120     0.438G     0.4601      0.319     0.8955         12        416: 100%|██████████| 662/662 [00:58<00:00, 11.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.18it/s]

                   all        331       1034      0.983       0.97      0.989      0.941



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     78/120     0.438G      0.457     0.3151     0.8976          7        416: 100%|██████████| 662/662 [00:58<00:00, 11.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.04it/s]

                   all        331       1034      0.972       0.98      0.989       0.94



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     79/120     0.438G     0.4339     0.3077     0.8873         21        416: 100%|██████████| 662/662 [00:57<00:00, 11.50it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.46it/s]

                   all        331       1034      0.981      0.972       0.99      0.939



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     80/120     0.438G     0.4328     0.3042     0.8924         12        416: 100%|██████████| 662/662 [00:56<00:00, 11.63it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.73it/s]

                   all        331       1034      0.983      0.968       0.99      0.944



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     81/120     0.438G     0.4443     0.3066     0.8906         10        416: 100%|██████████| 662/662 [00:57<00:00, 11.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.88it/s]

                   all        331       1034      0.976      0.976      0.989      0.938



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     82/120     0.438G     0.4365      0.303     0.8872          7        416: 100%|██████████| 662/662 [00:57<00:00, 11.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.90it/s]

                   all        331       1034      0.986       0.97       0.99      0.939



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     83/120     0.438G     0.4393     0.3079     0.8939         18        416: 100%|██████████| 662/662 [00:56<00:00, 11.64it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.22it/s]

                   all        331       1034      0.987      0.969      0.991      0.938



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     84/120     0.438G     0.4421     0.3072      0.896         12        416: 100%|██████████| 662/662 [00:57<00:00, 11.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.88it/s]

                   all        331       1034      0.976      0.976       0.99      0.941



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     85/120     0.438G     0.4292     0.2988     0.8857          6        416: 100%|██████████| 662/662 [00:57<00:00, 11.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.02it/s]

                   all        331       1034      0.981      0.972      0.989      0.944



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     86/120     0.438G     0.4238     0.2963     0.8878          5        416: 100%|██████████| 662/662 [00:57<00:00, 11.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.22it/s]

                   all        331       1034      0.981      0.969      0.989      0.944



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     87/120     0.438G     0.4239     0.2984      0.886         24        416: 100%|██████████| 662/662 [00:56<00:00, 11.66it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.21it/s]

                   all        331       1034      0.975       0.98      0.989      0.942



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     88/120     0.438G     0.4208     0.2936     0.8831         22        416: 100%|██████████| 662/662 [00:57<00:00, 11.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.80it/s]

                   all        331       1034      0.973      0.981       0.99      0.946



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     89/120     0.438G     0.4233     0.2984     0.8873         13        416: 100%|██████████| 662/662 [00:57<00:00, 11.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.75it/s]

                   all        331       1034      0.975      0.979      0.989      0.944



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     90/120     0.438G     0.4208     0.2961     0.8879         15        416: 100%|██████████| 662/662 [00:57<00:00, 11.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 24.01it/s]


                   all        331       1034      0.992      0.963       0.99      0.945

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     91/120     0.438G     0.4163     0.2903     0.8882         18        416: 100%|██████████| 662/662 [00:57<00:00, 11.61it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.54it/s]

                   all        331       1034      0.986      0.971       0.99      0.945



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     92/120     0.438G     0.4188      0.293     0.8851          8        416: 100%|██████████| 662/662 [00:57<00:00, 11.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.78it/s]

                   all        331       1034      0.993      0.963       0.99      0.944



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     93/120     0.438G     0.4172     0.2939     0.8871         15        416: 100%|██████████| 662/662 [00:57<00:00, 11.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.86it/s]

                   all        331       1034      0.979      0.975       0.99      0.942



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     94/120     0.438G     0.4066     0.2819     0.8839         19        416: 100%|██████████| 662/662 [00:57<00:00, 11.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.05it/s]

                   all        331       1034      0.984      0.971      0.989      0.949



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     95/120     0.438G     0.4112     0.2904     0.8808          9        416: 100%|██████████| 662/662 [00:57<00:00, 11.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.29it/s]

                   all        331       1034      0.993      0.962      0.989      0.945



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     96/120     0.438G     0.4211     0.3034     0.8894         10        416: 100%|██████████| 662/662 [00:57<00:00, 11.61it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.22it/s]

                   all        331       1034      0.977      0.976       0.99      0.946



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     97/120     0.438G     0.4073     0.2903     0.8846         11        416: 100%|██████████| 662/662 [00:57<00:00, 11.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.78it/s]

                   all        331       1034      0.992      0.964      0.991      0.949



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     98/120     0.438G       0.41     0.2895     0.8852          8        416: 100%|██████████| 662/662 [00:57<00:00, 11.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.17it/s]

                   all        331       1034      0.996      0.964       0.99       0.95



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     99/120     0.438G     0.4033     0.2809     0.8844         11        416: 100%|██████████| 662/662 [00:57<00:00, 11.58it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.01it/s]

                   all        331       1034      0.989      0.973       0.99      0.948



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    100/120     0.438G     0.4018     0.2814      0.879          3        416: 100%|██████████| 662/662 [00:57<00:00, 11.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.19it/s]

                   all        331       1034      0.991      0.971       0.99       0.95



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    101/120     0.438G     0.3956     0.2801     0.8788         10        416: 100%|██████████| 662/662 [00:57<00:00, 11.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.99it/s]

                   all        331       1034      0.989      0.972       0.99      0.949



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    102/120     0.438G      0.399     0.2813     0.8816         10        416: 100%|██████████| 662/662 [00:57<00:00, 11.54it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.99it/s]

                   all        331       1034      0.988      0.971       0.99      0.949



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    103/120     0.438G     0.3969     0.2756     0.8756         16        416: 100%|██████████| 662/662 [00:57<00:00, 11.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.17it/s]

                   all        331       1034      0.992      0.972       0.99      0.953



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    104/120     0.438G     0.3954     0.2762     0.8806         11        416: 100%|██████████| 662/662 [00:57<00:00, 11.58it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.33it/s]

                   all        331       1034      0.988       0.97       0.99      0.952



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    105/120     0.438G     0.3856     0.2759     0.8724         11        416: 100%|██████████| 662/662 [00:57<00:00, 11.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.02it/s]

                   all        331       1034      0.987      0.972       0.99      0.951



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    106/120     0.438G     0.3905     0.2774     0.8769         14        416: 100%|██████████| 662/662 [00:57<00:00, 11.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.14it/s]

                   all        331       1034       0.99      0.972       0.99      0.951



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    107/120     0.438G     0.3818     0.2705     0.8768          6        416: 100%|██████████| 662/662 [00:57<00:00, 11.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.37it/s]

                   all        331       1034      0.991      0.969       0.99      0.951



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    108/120     0.438G     0.3849     0.2734     0.8809          7        416: 100%|██████████| 662/662 [00:57<00:00, 11.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.10it/s]

                   all        331       1034      0.991      0.971       0.99      0.951



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    109/120     0.438G     0.3861     0.2729       0.88         13        416: 100%|██████████| 662/662 [00:57<00:00, 11.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.71it/s]

                   all        331       1034      0.991       0.97       0.99      0.951



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    110/120     0.438G     0.3801     0.2683     0.8758         20        416: 100%|██████████| 662/662 [00:57<00:00, 11.54it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.79it/s]

                   all        331       1034      0.992      0.968       0.99      0.952


Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    111/120     0.438G     0.3739     0.2526     0.8473          6        416: 100%|██████████| 662/662 [00:57<00:00, 11.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.06it/s]

                   all        331       1034      0.987      0.969       0.99       0.95



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    112/120     0.438G     0.3584     0.2443     0.8477          5        416: 100%|██████████| 662/662 [00:57<00:00, 11.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.00it/s]

                   all        331       1034      0.988       0.97       0.99       0.95



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    113/120     0.438G     0.3561      0.241     0.8464          2        416: 100%|██████████| 662/662 [00:57<00:00, 11.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.36it/s]

                   all        331       1034      0.985      0.976       0.99      0.951



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    114/120     0.438G     0.3508     0.2384     0.8428          8        416: 100%|██████████| 662/662 [00:57<00:00, 11.57it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 21.66it/s]

                   all        331       1034       0.99       0.97       0.99      0.953



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    115/120     0.438G     0.3424     0.2348     0.8444          5        416: 100%|██████████| 662/662 [00:57<00:00, 11.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.22it/s]

                   all        331       1034      0.992      0.972       0.99      0.953



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    116/120     0.438G     0.3416     0.2366     0.8413          6        416: 100%|██████████| 662/662 [00:56<00:00, 11.61it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 23.13it/s]

                   all        331       1034      0.989      0.973       0.99      0.953



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    117/120     0.438G     0.3424     0.2349     0.8409          8        416: 100%|██████████| 662/662 [00:57<00:00, 11.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.17it/s]

                   all        331       1034      0.991      0.974       0.99      0.953



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    118/120     0.438G     0.3351     0.2307     0.8426          6        416: 100%|██████████| 662/662 [00:57<00:00, 11.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.25it/s]

                   all        331       1034      0.982      0.979       0.99      0.952



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    119/120     0.438G     0.3343     0.2297     0.8416          8        416: 100%|██████████| 662/662 [00:57<00:00, 11.50it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.02it/s]

                   all        331       1034      0.983      0.977       0.99      0.952



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    120/120     0.438G     0.3356       0.23     0.8432          7        416: 100%|██████████| 662/662 [00:57<00:00, 11.55it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 22.02it/s]

                   all        331       1034      0.982      0.977       0.99      0.952



120 epochs completed in 2.066 hours.
Optimizer stripped from runs/detect/train2/weights/last.pt, 5.4MB
Optimizer stripped from runs/detect/train2/weights/best.pt, 5.4MB

Validating runs/detect/train2/weights/best.pt...
Ultralytics 8.3.141 🚀 Python-3.10.12 torch-2.7.0+cu126 CUDA:0 (NVIDIA GeForce MX550, 1864MiB)
YOLO11n summary (fused): 100 layers, 2,582,542 parameters, 0 gradients, 6.3 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 83/83 [00:03<00:00, 24.06it/s]


                   all        331       1034      0.992      0.972       0.99      0.953
                person        218        410      0.992       0.95      0.985      0.945
                teleco        331        624      0.992      0.993      0.995      0.962
Speed: 0.2ms preprocess, 7.5ms inference, 0.0ms loss, 0.6ms postprocess per image
Results saved to runs/detect/train2
✅  Trained checkpoint saved to: runs/detect/train2/weights/best.pt


# 🧠 YOLOv8 Training Metrics Cheatsheet

## 🔢 Loss Curves

| Metric            | Meaning                                                | Goal / Good Trend                |
|-------------------|--------------------------------------------------------|----------------------------------|
| `train/box_loss`  | Bounding box regression loss (training)                | ↓ Steady decline                 |
| `train/cls_loss`  | Classification loss (training)                         | ↓ Should fall and stabilize     |
| `train/dfl_loss`  | Distribution Focal Loss (for bounding box quality)     | ↓ Smooth drop                   |
| `val/box_loss`    | Box loss on validation set                             | ↓ Best indicator of generalization |
| `val/cls_loss`    | Classification loss on validation                      | ↓ Falling → Good model fit      |
| `val/dfl_loss`    | DFL loss on validation                                 | ↓ Matches train trend = no overfit |

---

## 📈 Detection Metrics

| Metric                | Description                                                 | Goal / Interpretation           |
|------------------------|-------------------------------------------------------------|----------------------------------|
| `metrics/precision(B)` | % of correct detections out of all predictions              | ↑ High = few false positives     |
| `metrics/recall(B)`    | % of actual objects detected                                | ↑ High = few missed detections   |
| `metrics/mAP50(B)`     | Accuracy of boxes at 50% IoU threshold                     | ↑ Main object detection metric   |
| `metrics/mAP50-95(B)`  | Average mAP over IoU 0.5 to 0.95 (stricter)                | ↑ Robust performance measure     |

---

## 🧪 Ideal Curve Patterns

- **Losses**: should steadily decrease.
- **Validation loss** rising while training loss decreases → possible **overfitting**.
- **Precision & Recall**: should increase and plateau near 1.0.
- **mAP**: aim for **>0.90 mAP50** and **>0.80 mAP50-95** for high-quality results.

---

## ⚠️ Tips

- 🧊 Use `half=True` for mixed precision training on supported GPUs.
- 🧠 Watch `val/box_loss` and `mAP50-95` to catch overfitting early.
- 🧪 If metrics plateau early, try increasing `epochs`, `imgsz`, or improving labels.
- 💾 Check `runs/train/.../results.png` for curve visualization.

---
